src/Workspaces/Core/Portable/Execution/Serializer.cs

```diff
@@ -88,6 +190,49 @@ public T Deserialize<T>(string kind, ObjectReader reader, CancellationToken canc
             }
         }
 
+        private string GetLanguageName(object value)
+        {
+            // for given object, we need to figure out which language the object belong to. 
+            // we can't blindly get language service since that will bring in language specific dlls.
+            foreach (var languageName in _workspaceServices.SupportedLanguages)
+            {
+                IOptionsSerializationService service;
+                if (_lazyLanguageSerializationService.TryGetValue(languageName, out service))
+                {
+                    if (service.Owns(value))
+                    {
+                        return languageName;
+                    }
+
+                    continue;
+                }
+
+                // this should be only reached once per language value actually belong to
+                var mefWorkspaceServices = _workspaceServices as MefWorkspaceServices;
+                if (mefWorkspaceServices != null)
+                {
+                    MefLanguageServices languageServices;
+                    if (!mefWorkspaceServices.TryGetLanguageServices(languageName, out languageServices))
+                    {
+                        // this is a bit fragile since it depends on implementation detail but there is no other way
+                        // to figure out which language a type belong to without loading other languages
```

Comment:
> I dont think we need that one in this PR though

IMO, this is one of those things that just won't get fixed unless someone just does the work.  IMO this is small enogh and valuable enough that we should just do it.   Brittle code like this is a continual tax on us.  Our system is already massively complex, and OOP increases that necessarily.  But i don't want to do things like this when there are perfectly good solutions at our disposal.